treeherder/model/sql/template_schema/project.sql.tmpl

```diff
@@ -377,12 +381,16 @@ DROP TABLE IF EXISTS `revision`;
 CREATE TABLE `revision` (
   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `revision` varchar(50) COLLATE utf8_bin NOT NULL,
+  `short_revision` varchar(12) COLLATE utf8_bin DEFAULT NULL,
+  `long_revision` varchar(40) COLLATE utf8_bin DEFAULT NULL,
   `author` varchar(150) COLLATE utf8_bin NOT NULL,
   `comments` text DEFAULT NULL,
   `repository_id` int(11) unsigned NOT NULL,
   `active_status` enum('active','onhold','deleted') COLLATE utf8_bin DEFAULT 'active',
   PRIMARY KEY (`id`),
   UNIQUE KEY `idx_revision` (`revision`, `repository_id`),
+  UNIQUE KEY `idx_short_revision` (`short_revision`, `repository_id`),
```

Comment:
I don't think we can make short_revision a unique key - it might not be unique due to collisions. Make an index instead? (Same for the result_set table)